### PR TITLE
interfaces: remove old API

### PR DESCRIPTION
**Pull request by stolowski**
This PR removes *Snippet methods of the old interfaces API to complete the transition to the spec-based API. This is based on PRs #3072 and #3077.

### CODE REVIEW

**Comment by stolowski** [comment]
@niemeyer it's mostly removing, but a few tests needed some changes, so still worth taking a look once prerequisites land.

File: interfaces/builtin/maliit_test.go [hunk 1]
```diff
@@ -114,9 +114,10 @@ func (s *MaliitInterfaceSuite) TestConnectedPlugSnippetUsesSlotLabelSome(c *C) {
 }
 
 func (s *MaliitInterfaceSuite) TestConnectedPlugSecComp(c *C) {
-	snippet, err := s.iface.ConnectedPlugSnippet(s.plug, s.slot, interfaces.SecuritySecComp)
+	seccompSpec := &seccomp.Specification{}
+	err := seccompSpec.AddConnectedPlug(s.iface, s.plug, s.slot)
```
Comment (zyga) [inline interfaces/builtin/maliit_test.go]: Ha, nice catch :-)

File: interfaces/core.go [hunk 2]
```diff
@@ -120,66 +120,6 @@ type Interface interface {
 	// SanitizeSlot checks if a slot is correct, altering if necessary.
 	SanitizeSlot(slot *Slot) error
 
-	// PermanentPlugSnippet returns the snippet of text for the given security
```
Comment (zyga) [inline interfaces/core.go]: Bye bye :-) Thank you!

File: interfaces/repo.go [hunk 3]
```diff
@@ -656,16 +656,6 @@ func (r *Repository) Interfaces() *Interfaces {
...
-// system that affect a given snap. The return value is indexed by app/hook
-// security tag within that snap.
-func (r *Repository) SecuritySnippetsForSnap(snapName string, securitySystem SecuritySystem) (map[string][][]byte, error) {
-	r.m.Lock()
-	defer r.m.Unlock()
-
-	return r.securitySnippetsForSnap(snapName, securitySystem)
-}
-
 func addSnippet(snapName, uniqueName string, apps map[string]*snap.AppInfo, hooks map[string]*snap.HookInfo, snippets map[string][][]byte, snippet []byte) {
```
Comment (zyga) [inline interfaces/repo.go]: Do we have an use for `addSnippet` after `securitySnippetsForSnap` is removed?

File: interfaces/repo_test.go [hunk 4]
```diff
@@ -1393,114 +1318,63 @@ func (s *RepositorySuite) TestSnapSpecification(c *C) {
...
-	c.Assert(err, IsNil)
-	c.Check(snippets, DeepEquals, map[string][][]byte{
-		"snap.snap-b.none.slot-b": {
-			[]byte(`static slot snippet`),
-			[]byte(`connection-specific slot snippet`),
-		},
-	})
-}
-
 func (s *RepositorySuite) TestSecuritySnippetsForSnapFailureWithConnectionSnippets(c *C) {
```
Comment (zyga) [inline interfaces/repo_test.go]: The test name refers to the old function name (`SecuritySnippetsForSnap`)

File: interfaces/repo_test.go [hunk 5]
```diff
@@ -1393,114 +1318,63 @@ func (s *RepositorySuite) TestSnapSpecification(c *C) {
...
 	c.Assert(err, ErrorMatches, "cannot compute snippet for consumer")
-	c.Check(snippets, IsNil)
+	c.Assert(spec, IsNil)
+
+	spec, err = repo.SnapSpecification(testSecurity, s.slot.Snap.Name())
+	c.Assert(err, ErrorMatches, "cannot compute snippet for provider")
+	c.Assert(spec, IsNil)
 }
 
 func (s *RepositorySuite) TestSecuritySnippetsForSnapFailureWithPermanentSnippets(c *C) {
```
Comment (zyga) [inline interfaces/repo_test.go]: Ditto test name

**Pull request merged**